cmd/terramate/cli/project.go

```diff
@@ -78,6 +78,15 @@ func (p *project) localDefaultBranchCommit() string {
 	return val
 }
 
+func (p *project) isGitFeaturesEnabled() bool {
+	return p.isRepo && p.hasCommits()
+}
+
+func (p *project) hasCommits() bool {
+	_, err := p.git.wrapper.RevParse("HEAD")
```

Comment:
A fully functioning repository must have `HEAD` symbolic-ref set up, otherwise the branch is in the `unborn` state.